src/v0/destinations/braze/util.js

```diff
@@ -299,9 +306,77 @@ const processDeduplication = (userStore, payload) => {
   return null;
 };
 
+const processBatch = (transformedEvents) => {
+  const { destination } = transformedEvents[0];
+  const attributesArray = [];
+  const eventsArray = [];
+  const purchaseArray = [];
+  const successMetadata = [];
+  const failureResponses = [];
+  for (const transformedEvent of transformedEvents) {
+    if (transformedEvent?.batchedRequest?.body?.JSON) {
+      const { attributes, events, purchases } = transformedEvent.batchedRequest.body.JSON;
+      if (Array.isArray(attributes)) {
+        attributesArray.push(...attributes);
+      }
+      if (Array.isArray(events)) {
+        eventsArray.push(...events);
+      }
+      if (Array.isArray(purchases)) {
+        purchaseArray.push(...purchases);
+      }
+      successMetadata.push(...transformedEvent.metadata);
+    } else if (!isHttpStatusSuccess(transformedEvent?.statusCode)) {
+      failureResponses.push(transformedEvent);
+    }
+  }
+  const attributeArrayChunks = _.chunk(attributesArray, 75);
+  const eventsArrayChunks = _.chunk(eventsArray, 75);
+  const purchaseArrayChunks = _.chunk(purchaseArray, 75);
+
+  const maxNumberOfRequest = Math.max(
+    attributeArrayChunks.length,
+    eventsArrayChunks.length,
+    eventsArrayChunks.length,
+  );
+  const responseArray = [];
+  for (let i = 0; i < maxNumberOfRequest; i += 1) {
+    const attributes = attributeArrayChunks[i];
+    const events = eventsArrayChunks[i];
+    const purchases = purchaseArrayChunks[i];
+    const response = defaultRequestConfig();
+    response.endpoint = getTrackEndPoint(getEndpointFromConfig(destination));
```

Comment:
[Q] Will `getEndpointFromConfig` change wrt the event ? I mean can we call this method outside the loop and make use of the value from there ?!

Reply:
Yep.handled